src/openfluid/ui/waresdev/WareBuildInfoWidget.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+
+  This file is part of OpenFLUID software
+  Copyright(c) 2007, INRA - Montpellier SupAgro
+
+
+ == GNU General Public License Usage ==
+
+  OpenFLUID is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OpenFLUID is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OpenFLUID. If not, see <http://www.gnu.org/licenses/>.
+
+
+ == Other Usage ==
+
+  Other Usage means a use of OpenFLUID that is inconsistent with the GPL
+  license, and requires a written agreement between You and INRA.
+  Licensees for Other Usage of OpenFLUID may use this file in accordance
+  with the terms contained in the written agreement between You and INRA.
+  
+*/
+
+
+/**
+  @file WareBuildInfoWidget.cpp
+
+  @author Armel THONI <armel.thoni@inrae.fr>
+*/
+
+
+#include <openfluid/ui/waresdev/WareBuildInfoWidget.hpp>
+#include <openfluid/base/RunContextManager.hpp>
+#include <openfluid/ui/common/UIHelpers.hpp>
+
+#include "ui_WareBuildInfoWidget.h"
+
+
+namespace openfluid { namespace ui { namespace waresdev {
+
+
+WareBuildInfoWidget::WareBuildInfoWidget(QWidget* Parent):
+  QWidget(Parent), ui(new Ui::WareBuildInfoWidget)
+{
+  ui->setupUi(this);
+
+  ui->buildOptionsButton->setText("");
+  ui->buildOptionsButton->setIcon(openfluid::ui::common::getIcon("settings","/ui/common"));
+
+  connect(ui->buildOptionsButton, SIGNAL(clicked()), this, SIGNAL(settingsButtonClicked()));
+  connect(ui->jobStatus, SIGNAL(scrolled(bool)), this, SIGNAL(jobsScrolled(bool)));
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+WareBuildInfoWidget::~WareBuildInfoWidget()
+{
+  delete ui;
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+openfluid::ui::common::ActionLabel* WareBuildInfoWidget::getLabel(std::string LabelName)
+{
+  if (LabelName == "Installed")
+  {
+    return ui->buildStatus;
+  }
+  else if (LabelName == "Mode")
+  {
+    return ui->configStatus;
+  }
+  else if (LabelName == "Jobs")
+  {
+    return ui->jobStatus;
+  }
+}
```

Comment:
Unknown (dangerous) behaviour if given LabelName is none of the expected ones :boom: 
See also comment in the corresponding hpp file